scripts/cbmc_json_parser.py

```diff
@@ -95,9 +96,37 @@ def __init__(self, source_location={}):
         self.column = source_location.get("column", None)
         self.line = source_location.get("line", None)
 
+    def filepath(self):
+        """ Return a more user friendly path.
+
+        - If the path is inside the current directory, return a relative path.
+        - If not but the path is in the ${HOME} directory, return relative to ${HOME}
+        - Otherwise, return the path as is.
+        """
+        if not self.filename:
+            return None
+
+        if not path.exists(self.filename):
```

Comment:
Is this necessary for what comes after? Checking if the file exists can be expensive (since it involves an OS/filesystem operation) which can slow things down quite a bit if we're checking the path for every check.

Reply:
I thought that this fixed an issue that I saw before, but I couldn't reproduce it. So I'm removing this. :)

I added a few tests to hopefully ensure that I didn't miss anything.